api/src/main/java/org/openmrs/module/eptsreports/reporting/library/dimensions/EptsCommonDimension.java

```diff
@@ -205,4 +214,110 @@ public CohortDefinitionDimension maternityDimension() {
+
+    dim.addCohortDefinition(
+        "LTFU",
+        EptsReportUtils.map(
+            genericCohortQueries.getLostToFollowUpPatients(),
+            "endDate=${endDate},location=${location}"));
+
+    dim.addCohortDefinition(
+        "TOP",
+        EptsReportUtils.map(
+            genericCohortQueries.getPatientsBasedOnPatientStates(
+                hivMetadata.getARTProgram().getProgramId(),
+                hivMetadata
+                    .getTransferredOutToAnotherHealthFacilityWorkflowState()
+                    .getProgramWorkflowStateId()),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "STP",
+        EptsReportUtils.map(
+            genericCohortQueries.getPatientsBasedOnPatientStates(
+                hivMetadata.getARTProgram().getProgramId(),
+                hivMetadata.getSuspendedTreatmentWorkflowState().getProgramWorkflowStateId()),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "adults",
+        EptsReportUtils.map(
+            txPvlsCohortQueries.findPatientsBetweenAgeBracketsInYears(15, 200),
+            "endDate=${endDate},location=${location}"));
+    return dim;
+  }
+
+  /**
+   * Get the dimensions based on the patient states for ERI-4 months
+   *
+   * @return CohortDefinitionDimension
+   */
+  public CohortDefinitionDimension getEri2MonthsDimension() {
+    CohortDefinitionDimension dim = new CohortDefinitionDimension();
+    dim.addParameter(new Parameter("endDate", "End Date", Date.class));
+    dim.addParameter(new Parameter("location", "location", Location.class));
+    dim.setName("Get patients dimensions for Eri2Months");
+    dim.addCohortDefinition(
+        "IART",
+        EptsReportUtils.map(
+            eri2MonthsCohortQueries.getAllPatientsWhoInitiatedArt(),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "DNPUD",
+        EptsReportUtils.map(
+            eri2MonthsCohortQueries.getPatientsWhoDidNotPickDrugsOnTheirSecondVisit(),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "PUD",
+        EptsReportUtils.map(
+            eri2MonthsCohortQueries.getPatientsWhoPickedUpDrugsOnTheirSecondVisit(),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "DP",
+        EptsReportUtils.map(
+            genericCohortQueries.getDeceasedPatients(), "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "TOP",
+        EptsReportUtils.map(
+            genericCohortQueries.getPatientsBasedOnPatientStates(
+                hivMetadata.getARTProgram().getProgramId(),
+                hivMetadata
+                    .getTransferredOutToAnotherHealthFacilityWorkflowState()
+                    .getProgramWorkflowStateId()),
+            "endDate=${endDate},location=${location}"));
+    dim.addCohortDefinition(
+        "STP",
```

Comment:
Same here: This looks like it is: `A and B and Stopped/Suspended`

but specs is looking for: `A and B and C and Stopped/Suspended`

Looks like the query C is missing.

Reply:
Yah, that is correct, will update that, the C is missing in that dimension

Reply:
Done